lockfile/src/parsers/go_sum.rs

```diff
@@ -0,0 +1,53 @@
+use std::ops::Not;
+
+use nom::bytes::complete::take_until;
+use nom::character::complete::space0;
+use nom::combinator::recognize;
+use phylum_types::types::package::{PackageDescriptor, PackageType};
+
+use super::Result;
+
+pub fn parse(input: &str) -> Result<&str, Vec<PackageDescriptor>> {
+    let pkgs = input.lines().filter_map(package).collect::<Vec<_>>();
+    Ok((input, pkgs))
+}
```

Comment:
This function never returns an error...

We've had problems in the past with other parsers being too permissive and handling files that they shouldn't. I definitely think that will be a problem here.

Reply:
We need to be especially careful that no other lockfile accidentally produces go packages, since our `auto` parser would then get confused.

Reply:
I reworked this to error when encountering unexpected input – now if it fails to find lines with three strings separated by spaces then it will fail. Hopefully that will give us more actionable insight vs before when it would have just returned no descriptors